lua/entities/kleiner/init.lua

```diff
@@ -0,0 +1,970 @@
+		
+		timer.Simple(2.4,function()
+			if(IsValid(self) )then
+			self:AddGestureSequence(self:LookupSequence("blowclipboardgesture"))
+			end
+		end)
+
+		self.Suicidal = true
+		
+		local pos,ang = self:GetBonePosition(self:LookupBone("ValveBiped.Bip01_L_Hand") or 0)
+		pos = pos + ang:Right() * 2
+		pos = pos + ang:Forward() * 4
+		local gren = self:SpawnGrenade()
+		gren:SetPos(pos)
+		gren:SetAngles(ang)
+		gren:SetParent(self,10)
+		self.Grenade1 = gren
+		local pos,ang = self:GetBonePosition(self:LookupBone("ValveBiped.Bip01_R_Hand") or 0)
+		pos = pos + ang:Right() * 2
+		pos = pos + ang:Forward() * 4
+		ang:RotateAroundAxis(ang:Right(),180)
+		local gren = self:SpawnGrenade()
+		gren:SetPos(pos)
+		gren:SetAngles(ang)
+		gren:SetParent(self,11)
+		self.Grenade2 = gren
+		timer.Simple(1,function()
+			if(IsValid(self) )then
+			if(IsValid(self:GetTarget()) )then self:SetTargetViolence(self:GetTarget(),math.floor(self:GetTargetViolence(self:GetTarget())/1.15)) end
+				if(IsValid(self.Grenade1) and IsValid(self.Grenade1:GetParent()))then self.Grenade1.DamageTriggered = true self.Grenade1:Fire("SetTimer",3) end
+				if(IsValid(self.Grenade2) and IsValid(self.Grenade2:GetParent()))then self.Grenade1.DamageTriggered = true self.Grenade2:Fire("SetTimer",3) end
+			end
+		end)
+	end
+end
+
+function ENT:SpawnGrenade()
+	local gren = ents.Create("npc_grenade_frag")
+	gren:Spawn()
+	gren:Activate()
+	gren:SetMoveType(MOVETYPE_NONE)
+	gren:PhysicsDestroy()
+	gren:SetCollisionGroup(COLLISION_GROUP_WORLD)
+	gren:SetOwner(self)
+	gren.Owner = self
+	return gren
+end
+
+function ENT:DropGrenades()
+	if(IsValid(self.Grenade1))then 
+		local gren = self.Grenade1
+		gren:SetParent() 
+		gren:PhysicsInit(MOVETYPE_VPHYSICS) 
+		gren:GetPhysicsObject():Wake() 
+		timer.Simple(60*15,function()
+			if(IsValid(gren))then
+			gren:Remove()
+			end
+		end)
+	end
+	if(IsValid(self.Grenade2))then 
+		local gren = self.Grenade2
+		gren:SetParent() 
+		gren:PhysicsInit(MOVETYPE_VPHYSICS) 
+		gren:GetPhysicsObject():Wake() 
+		timer.Simple(60*15,function()
+			if(IsValid(gren))then
+			gren:Remove()
+			end
+		end)
+	end
+end
+
+
+if(SERVER)then
+KLEINER_BULLIES = KLEINER_BULLIES or {}
+end
+ 
+ 
+function ENT:PosInRange(pos)
+	local heightlimit = self.TargetHeight*1
+	local horizontalrange = (self:GetPos()*Vector(1,1,0)):Distance(pos*Vector(1,1,0))
+	local heightdiff = math.abs((self:GetPos() - pos).z)
+	if(horizontalrange < self.TargetHeightInnerRadius)then
+		heightlimit = self.TargetHeightInner
+	end
+	return horizontalrange <= self.LoseTargetDist and heightdiff <= heightlimit
+end
+ 
+function ENT:CanBeTarget(ent)
+	if(!IsValid(ent))then return false end
+	if(ent == self)then return false end
+
+	if(ent.InVehicle and ent:InVehicle())then return false end
+	if(ent.Alive and !ent:Alive())then return false end
+	if(ent:GetMoveType() == MOVETYPE_FLY)then return false end
+	if(player.IsAFK and player:IsAFK())then return false end
```

Comment:
mixing `ent` and `player`. Do you expect this to target entities other than players? If not, don't add extra checks

Reply:
they were designed originally to target eachother, but i figured it would be a waste of serverside logic to have them behave away from players. i will remove them